Spacing/Dimensions Supports: Separate spacing from dimensions for compatibility purposes

aaronrobertshaw opened the pull request:
Related: https://github.com/WordPress/gutenberg/pull/32392

## Description

This PR addresses an issue of compatibility with core created by the renaming of spacing to dimensions support in [#32392](https://github.com/WordPress/gutenberg/pull/32392#discussion_r687501190).

- The previous version of `spacing.php` has been restored
- The `dimensions.php` server-side hook remains in place ready to accept [height](https://github.com/WordPress/gutenberg/pull/32499) and [width](https://github.com/WordPress/gutenberg/pull/32502) support
- No changes to the block.json or theme.json support keys have been made
   - `spacing` contains `padding` and `margin` support
       - block.json and theme.json both use the stabilised `spacing` key.
   - `dimensions` will contain `height`, `width`, and possibly min/max versions of those.
       - block.json uses an `__experimentalDimensions` key
       - theme.json uses a `dimensions` key

## How has this been tested?

- `npm run test-unit:watch packages/block-editor/src/hooks/test/style.js`
- `npm run wp-env run phpunit 'phpunit -c /var/www/html/wp-content/plugins/gutenberg/phpunit.xml.dist --verbose --filter stylesheet /var/www/html/wp-content/plugins/gutenberg/phpunit/class-wp-theme-json-test.php'`
- Manually by:
    - opting into both padding and margin support under spacing for the group block
    - opting into both padding and margin support for the dynamic site tagline block
    - confirming dimensions hook is in place but takes no effect for the moment until the [height](https://github.com/WordPress/gutenberg/pull/32499) and [width](https://github.com/WordPress/gutenberg/pull/32502) supports add their application of styles within it

## Types of changes
Bug fix.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/javascript/ -->
- [x] My code follows the accessibility standards. <!-- Guidelines: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/accessibility/ -->
- [x] I've tested my changes with keyboard and screen readers. <!-- Instructions: https://github.com/WordPress/gutenberg/blob/HEAD/docs/contributors/accessibility-testing.md -->
- [x] My code has proper inline documentation. <!-- Guidelines: https://developer.wordpress.org/coding-standards/inline-documentation-standards/javascript/ -->
- [x] I've included developer documentation if appropriate. <!-- Handbook: https://developer.wordpress.org/block-editor/ -->
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal). <!-- React Native mobile Gutenberg guidelines: https://github.com/WordPress/gutenberg/blob/HEAD/docs/contributors/code/native-mobile.md -->

aaronrobertshaw committed:
Restore old spacing.php and remove spacing from dimensions hook

andrewserong reviewed (approved):
Thanks for following up on this @aaronrobertshaw!

✅ Confirmed that the diff between the old `spacing.php` and the re-introduced `spacing.php` is just the doc comment added at the top of the file
✅ Minimalist `dimensions.php` file looks good and ready for us to add in height and width supports 👍 
✅ Smoke tested that the existing spacing supports still work as expected (padding and margin in Group block, and the Button block's padding which skips serialization)

Also, IMO it feels like it'll be useful having the width and height controls separately in the dimensions support as that way it'll be possible to have slightly more granular control over skipping serialization between height/width and padding/margin 👍

andrewserong commented on lib/block-supports/dimensions.php:
```diff
@@ -38,79 +42,39 @@ function gutenberg_register_dimensions_support( $block_type ) {
- * @return array Block spacing CSS classes and inline styles.
+ * @return array Block dimensions CSS classes and inline styles.
  */
 function gutenberg_apply_dimensions_support( $block_type, $block_attributes ) {
```
PHP linting is complaining that `$block_attributes` is unused, but aside from that this LGTM!

aaronrobertshaw replied:
Thanks, I've pushed a commit temporarily disabling this rule. It will be removed in the [height block support PR](https://github.com/WordPress/gutenberg/pull/32499) where this parameter will actually be used.

aaronrobertshaw commented:
> Also, IMO it feels like it'll be useful having the width and height controls separately in the dimensions support as that way it'll be possible to have slightly more granular control over skipping serialization between height/width and padding/margin 

That's a good point. Which I think might tip the decision in favour of this approach rather than unifying the support keys.

aaronrobertshaw committed:
Temporarily disable the unused parameter linting

This ignore will be removed once height block support is added as the parameter will be used then.

ramonjd reviewed (approved):
Bit late, but also tested.

✔️ Controls appear as expected. 
✔️ Changes reflected in the frontend
✔️ Tests pass

Thank you!

oandregal reviewed (approved):
Haven't tested but the code changes look good.
